velox/functions/prestosql/tests/ArrayAllMatchTest.cpp

```diff
@@ -0,0 +1,184 @@
+  // Create an array of array vector using above base vector using offsets.
+  // [
+  //  [[1, 2, 3]],
+  //  [[2, 2], [3, 3], [4, 4], [5, 5]],
+  //  [[]]
+  // ]
+  auto arrayOfArrays = makeArrayVector({0, 1, 5}, baseVector);
+  auto result = evaluate<SimpleVector<bool>>(
+      "all_match(c0, x -> (cardinality(x) > 0))",
+      makeRowVector({arrayOfArrays}));
+  auto expectedResult = makeNullableFlatVector<bool>({true, true, false});
+  assertEqualVectors(expectedResult, result);
+
+  // Create an array of array vector using above base vector using offsets.
+  // [
+  //  [[1, 2, 3]],  cardinalities is 3
+  //  [[2, 2], [3, 3], [4, 4], [5, 5]], all cardinalities is 2
+  //  [[]],
+  //  null
+  // ]
+  arrayOfArrays = makeArrayVector({0, 1, 5, 6}, baseVector, {3});
+  result = evaluate<SimpleVector<bool>>(
+      "all_match(c0, x -> (cardinality(x) > 2))",
+      makeRowVector({arrayOfArrays}));
+  expectedResult =
+      makeNullableFlatVector<bool>({true, false, false, std::nullopt});
+  assertEqualVectors(expectedResult, result);
+}
+
+TEST_F(ArrayAllMatchTest, bigints) {
+  auto input = makeNullableArrayVector<int64_t>(
+      {{},
+       {2},
+       {std::numeric_limits<int64_t>::max()},
+       {std::numeric_limits<int64_t>::min()},
+       {std::nullopt, std::nullopt}, // return null if all is null
+       {2,
+        std::nullopt}, // return null if one or more is null and others matched
+       {1, std::nullopt, 2}}); // return false if one is not matched
+  auto result = evaluate<SimpleVector<bool>>(
+      "all_match(c0, x -> (x % 2 = 0))", makeRowVector({input}));
+
+  auto expectedResult = makeNullableFlatVector<bool>(
+      {true, true, false, true, std::nullopt, std::nullopt, false});
+  assertEqualVectors(expectedResult, result);
+}
+
+TEST_F(ArrayAllMatchTest, strings) {
+  auto input = makeNullableArrayVector<StringView>(
+      {{}, {"abc"}, {"ab", "abc"}, {std::nullopt}});
+  auto result = evaluate<SimpleVector<bool>>(
+      "all_match(c0, x -> (x = 'abc'))", makeRowVector({input}));
+
+  auto expectedResult =
+      makeNullableFlatVector<bool>({true, true, false, std::nullopt});
+  assertEqualVectors(expectedResult, result);
+}
+
+TEST_F(ArrayAllMatchTest, doubles) {
+  auto input =
+      makeNullableArrayVector<double>({{}, {1.2}, {3.0, 0}, {std::nullopt}});
+  auto result = evaluate<SimpleVector<bool>>(
+      "all_match(c0, x -> (x > 1.1))", makeRowVector({input}));
+
+  auto expectedResult =
+      makeNullableFlatVector<bool>({true, true, false, std::nullopt});
+  assertEqualVectors(expectedResult, result);
+}
+
+TEST_F(ArrayAllMatchTest, errors) {
+  // No throw and return false if there are unmatched elements except nulls
+  auto input = makeRowVector({makeNullableArrayVector<int8_t>(
+      {{0, 2, 0, 5, 0}, {5, std::nullopt, 0}})});
+  auto result =
+      evaluate<SimpleVector<bool>>("all_match(c0, x -> ((10 / x) > 2))", input);
```

Comment:
nit: can drop template parameters here and in other similar places as well

Reply:
Done